specs/FileTypePolicy.md

```diff
@@ -0,0 +1,225 @@
+File Type Policy API
+===
+
+# Background
+When saving a file with original SaveFilePicker, a security alert might be
+prompted, because the browser applies the [file type policies](https://learn.microsoft.com/en-us/deployedge/microsoft-edge-security-downloads-interruptions#file-types-requiring-a-gesture)
+to protect end users. However, in a WebView2 build App, when end users try
+to save a file with a certain file extension, they usually can trust the
+host App, domain and file extension. So, we provide the App developers the
+File Type Policy API to manage the file type policies dynamically.
+
+We'd appreciate your feedback.
+
+# Description
+
+We proposed the `CoreWebView2.FileTypePolicyChecking` event. You can register
+this event to get the file path, file extension and domain uri information,
+when end users try to save a file from your App. Then you can apply your own
+rules to allow save the file with, or without a default warning dialog;
+to cancel the saving; and even to create your own UI to manage runtime 
+file type policies.
+
+# Examples
+## Win32 C++ 
+This example shows suppressing file type policy, security dialog, and 
+allow to save the file directly. It also blocks saving the exe file.
+The sample code will register the event with custom rules.
+```c++
+bool ScenarioFileTypePolicyStaging::AddCustomFileTypePolicies()
+{
+    if (!m_webView2Staging25)
+        return false;
+    // Register a handler for the `FileTypePolicyChecking` event.
+    m_webView2Staging25->add_FileTypePolicyChecking(
+        Callback<ICoreWebView2StagingFileTypePolicyCheckingEventHandler>(
+            [this](
+                ICoreWebView2* sender,
+                ICoreWebView2StagingFileTypePolicyCheckingEventArgs* args) -> HRESULT
+            {
+                // Get the file extension for file to be saved.
+                // And create your own rules of file type policy.
+                LPWSTR extension;
+                CHECK_FAILURE(args->get_FileExtension(&extension));
+                if (lstrcmpW(extension, L"eml") == 0)
```

Comment:
If the file path is ...file.EmL will the FileExtension be 'eml' or 'EmL'? If sample code needs to check for a file extension does it need to do a case insensitive string comparison?